nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -161,11 +144,18 @@ class AckermannMotionModel : public MotionModel
    */
   void applyConstraints(models::ControlSequence & control_sequence) override
   {
-    auto & vx = control_sequence.vx;
-    auto & wz = control_sequence.wz;
+    const auto vx_ptr = control_sequence.vx.data();
+    auto wz_ptr = control_sequence.wz.data();
+    int steps = control_sequence.vx.size();
+    for(int i = 0; i < steps; i++)
+    {
+      float && wz_constrained = std::abs(*(vx_ptr + i) / min_turning_r_);
```

Comment:
Why all `i+1`? You could start with `i=1` - though I think you still want to apply the first index, no? That's what we did before

Reply:
Sorry, this was a dyslexic moment, I read `i` as `1`. Carry on...

Performance Q: worth storing the pointer so we can do a `curr_vx_ptr++` operation to march to the next one from the current one instead of starting with `vx_ptr` (start) `+ i` (iterating to it). The caching and compiler might solve this for it, but worth a try to see if there's a performance difference.